shuup/admin/template_helpers/shuup_admin.py

```diff
@@ -46,15 +47,29 @@ def get_menu_entry_categories(context):
     return menu.get_menu_entry_categories(request=context["request"])
 
 
-def is_menu_category_active(category, target_url):
+@lru_cache()
+def get_all_target_urls(context, target_url, breadcrumbs):
```

Comment:
can you pass the request here instead of the entire context? If I am not wrong, I already faced some issues when caching context with lru_cache as some weird content couldn't be serialized by the cache.